File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()) return 0;
        
        map<char,int>sc,tc;
        
        for(int i=0;i<s.size();i++){
            sc[s[i]]++;
            tc[t[i]]++;
        }
        //map<char,int>::iterator x=sc.begin();
        for(auto x:sc){
            if(sc[x.first]!=tc[x.first])return 0;
        }
          return 1;
    }
  
};

File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        //watch out both anagrams words are same of size
        if(s.length()!=t.length())return false;
        
        char []sChar=s.toCharArray();
        char []tChar=t.toCharArray();

        Arrays.sort(sChar);
        Arrays.sort(tChar);
        return new String(sChar).equals(new String(tChar));
    
    }
} 